airflow/providers/cncf/kubernetes/decorators/kubernetes.py

```diff
@@ -0,0 +1,149 @@
+        Defaults to False.
+    """
+
+    template_fields: Sequence[str] = ('op_args', 'op_kwargs')
+
+    # since we won't mutate the arguments, we should just do the shallow copy
+    # there are some cases we can't deepcopy the objects (e.g protobuf).
+    shallow_copy_attrs: Sequence[str] = ('python_callable',)
+
+    def __init__(
+        self,
+        **kwargs,
+    ) -> None:
+        self.pickling_library = pickle
+
+        # Image, name and namespace are all required.
+        if 'image' not in kwargs:
+            kwargs['image'] = 'python:3.8-slim-buster'
+
+        if 'name' not in kwargs:
+            kwargs['name'] = f'k8s_airflow_pod_{uuid.uuid4().hex}'
+
+        if 'namespace' not in kwargs:
+            kwargs['namespace'] = 'default'
+
+        super().__init__(**kwargs)
+
+    def execute(self, context: 'Context'):
+
+        with TemporaryDirectory(prefix='venv') as tmp_dir:
+            script_filename = os.path.join(tmp_dir, 'script.py')
+            py_source = self._get_python_source()
+
+            jinja_context = dict(
+                op_args=self.op_args,
+                op_kwargs=self.op_kwargs,
+                pickling_library=self.pickling_library.__name__,
+                python_callable=self.python_callable.__name__,
+                python_callable_source=py_source,
+                string_args_global=False,
+            )
+            write_python_script(
+                jinja_context=jinja_context,
+                filename=script_filename,
+                template_file='python_kubernetes_script.jinja2',
+            )
+
+            self.env_vars.append(
+                k8s.V1EnvVar(name="__PYTHON_SCRIPT", value=_read_file_contents(script_filename))
+            )
+
+            self.cmds.append("bash")
+
+            self.arguments.append("-cx")
+            self.arguments.append(
+                f'{_generate_decode_command("__PYTHON_SCRIPT", "/tmp/script.py")} && python /tmp/script.py'
+            )
+
+            return super().execute(context)
+
+    def _get_python_source(self):
+        raw_source = inspect.getsource(self.python_callable)
+        res = dedent(raw_source)
+        res = remove_task_decorator(res, "@task.kubernetes")
+        return res
+
+
+T = TypeVar("T", bound=Callable)
+
+
+def kubernetes_task(
+    python_callable: Optional[Callable] = None, multiple_outputs: Optional[bool] = None, **kwargs
+) -> TaskDecorator:
+    """
+    Kubernetes operator decorator. Wraps a function to be executed in K8s using KubernetesPodOperator.
+    Also accepts any argument that DockerOperator will via ``kwargs``. Can be reused in a single DAG.
+
+    :param python_callable: Function to decorate
+    :param multiple_outputs: if set, function return value will be
+        unrolled to multiple XCom values. List/Tuples will unroll to xcom values
+        with index as key. Dict will unroll to xcom values with keys as XCom keys.
+        Defaults to False.
```

Comment:
```suggestion
    :param multiple_outputs: if set, function return value will be
        unrolled to multiple XCom values. Dict will unroll to xcom values with keys as XCom keys.
        Defaults to False.
```
Same comment here on types handled with `multiple_outputs`.